src/components/referrals/ReferralsComponent.vue

```diff
@@ -0,0 +1,201 @@
+                        disabled
+                        value="">
+                        Wybierz typ badania:
+                    </option>
+                    <option
+                        v-for="research in typeOfMedicalExamination"
+                        :key="research.keyName">
+                        {{ research.typeName }}
+                    </option>
+                </select>
+            </div>
+
+            <div class="inputBox">
+                <label for="dateOfResource"
+                    >Wybierz dzień wystawienia skierowania:</label
+                >
+                <input
+                    id="dateOfResource"
+                    v-model="researchDate"
+                    type="date" />
+            </div>
+
+            <div class="actionsAndInfo">
+                <button-component
+                    :disabled="disableAddTest"
+                    wide
+                    >Zapisz skierowanie</button-component
+                >
+                <bouncing-balls-component :visible="isLoading" />
+                <error-info :message="error" />
+                <span
+                    v-show="successFlag"
+                    class="dataSend"
+                    >Skierowanie zostało przesłane pomyślnie</span
+                >
+            </div>
+        </form>
+    </div>
+</template>
+
+<script setup>
+    import ButtonComponent from '../ButtonComponent.vue';
+    import BouncingBallsComponent from '../BouncingBallsComponent.vue';
+    import ErrorInfo from '../ErrorInfo.vue';
+    import getDoctorsArray from '../../composables/getDoctor.js';
+    import typeOfSpecialization from '../../utils/listOfSpecialization.js';
+    import typeOfMedicalExamination from '../../utils/listOfTests.js';
+    import { computed, ref } from 'vue';
+    import { timestamp } from '../../firebase/config.js';
+    import useUserStore from '../../stores/userStore.js';
+    import useCollection from '../../composables/useCollections.js';
+    const { isLoading, addDoc, error } = useCollection('listOfReferrals');
+    const dataDoctor = ref(null);
+    const specializations = ref(null);
+    const typeOfTestPeople = ref(null);
+    const researchDate = ref(null);
+    const successFlag = ref(false);
+    const userStore = useUserStore();
+    const disableAddTest = computed(() => !dateFormat.value);
+    const dateFormat = computed(
+        () =>
+            researchDate.value &&
+            researchDate.value.split('-').reverse().join('-')
+    );
+    const addTest = async () => {
+        try {
+            await addDoc({
+                userId: userStore.getUserId,
+                date: dateFormat.value,
+                doctorData: dataDoctor.value,
+                specializationType: specializations.value,
+                testType: typeOfTestPeople.value,
+                createdAt: timestamp(),
+            });
+            if (error.value) throw new Error();
+            isLoading.value = false;
```

Comment:
why you manually reset variable which is ref that belongs to fucntionality of useCollection hook? Look at that. If you change state isLoading ref in useCollection hook, you cannot overwrite it in this component. firstly other function control behaviour of this variable secondly: loading effect (I mean bouncing balls) will never be displayed :/

Reply:
ik, i've added missing variables

Reply:
No you shouldnt add new variables. Rather delete isloading changng state if in useCollection hook isloading is changing. See yourself in definition